node-it/src/test/scala/com/wavesplatform/it/asset/InvokeReissueBurnAssetSuite.scala

```diff
@@ -0,0 +1,333 @@
+
+    for (data <- Seq(simpleNonreissuableAsset, simpleReissuableAsset, nftAsset)) s"${data.t} could be fully burned" in {
+      val acc     = createDapp(script(data))
+      val fee     = invocationCost(if (isNft(data)) 0 else 1)
+      val txIssue = issue(acc, method, data, fee)
+      val assetId = validateIssuedAssets(acc, txIssue, data, method = method)
+
+      burn(acc, method, assetId, data.q)
+
+      sender.assetsDetails(assetId).quantity shouldBe 0
+      sender.assertAssetBalance(acc, assetId, 0)
+    }
+
+    "Reissuable asset could be reissued" in {
+      val acc               = createDapp(script(simpleReissuableAsset))
+      val txIssue           = issue(acc, method, simpleReissuableAsset, invocationCost(1))
+      val assetId           = validateIssuedAssets(acc, txIssue, simpleReissuableAsset, method = method)
+      val initialQuantity   = simpleReissuableAsset.q
+      val addedQuantity     = 100500
+      val initialReissuable = simpleReissuableAsset.r
+
+      reissue(acc, method, assetId, addedQuantity, !initialReissuable)
+
+      sender.assetsDetails(assetId).reissuable shouldBe !initialReissuable
+      sender.assetsDetails(assetId).quantity shouldBe initialQuantity + addedQuantity
+      sender.assertAssetBalance(acc, assetId, initialQuantity + addedQuantity)
+    }
+
+    "Non-reissuable asset could not be reissued" in {
+      val acc     = createDapp(script(simpleNonreissuableAsset))
+      val txIssue = issue(acc, method, simpleNonreissuableAsset, invocationCost(1))
+      val assetId = validateIssuedAssets(acc, txIssue, simpleNonreissuableAsset, method = method)
+      assertBadRequestAndMessage(
+        reissue(acc, method, assetId, 100500, false),
+        "Asset is not reissuable"
+      )
+    }
+  }
+
+  "Restrictions in @Callable" - {
+    val method = "@Callable"
+
+    "Issue two identical with the same nonce should not produce an error " in {
+      val acc      = createDapp(script(simpleNonreissuableAsset))
+      val txIssue1 = issue(acc, method, simpleNonreissuableAsset, invocationCost(1))
+      val txIssue2 = issue(acc, method, simpleNonreissuableAsset, invocationCost(1))
+      validateIssuedAssets(acc, txIssue1, simpleNonreissuableAsset, method = method)
+      validateIssuedAssets(acc, txIssue2, simpleNonreissuableAsset, method = method)
+    }
+
+    "Reissuing more than Long.Max should produce an error" in {
+      val acc             = createDapp(script(simpleReissuableAsset))
+      val txIssue         = issue(acc, method, simpleReissuableAsset, invocationCost(1))
+      val assetId         = validateIssuedAssets(acc, txIssue, simpleReissuableAsset, method = method)
+      val initialQuantity = simpleReissuableAsset.q
+
+      assertBadRequestAndMessage(
+        reissue(acc, method, assetId, Long.MaxValue - initialQuantity + 1, true),
+        "State check failed. Reason: Asset total value overflow"
+      )
+    }
+
+    "Burning more than current asset count should produce an error" in {
```

Comment:
Try to transfer part of tokens to another account before burning here.

Reply:
there is a bug here (SC-576)

Reply:
+